contracts/upgradeable_contracts/amb_erc677_to_erc677/ForeignStakeTokenMediator.sol

```diff
@@ -0,0 +1,60 @@
+pragma solidity 0.4.24;
+
+import "./BasicStakeTokenMediator.sol";
+import "../../interfaces/IBurnableMintableERC677Token.sol";
+
+contract ForeignStakeTokenMediator is BasicStakeTokenMediator {
+    /**
+     * @dev Executes action on the request to withdraw tokens relayed from the other network
+     * @param _recipient address of tokens receiver
+     * @param _value amount of bridged tokens
+     */
+    function executeActionOnBridgedTokens(address _recipient, uint256 _value) internal {
+        uint256 value = _value.div(10**decimalShift());
+        _transferWithOptionalMint(_recipient, value);
+    }
+
+    /**
+     * @dev Executes action on deposit of bridged tokens
+     * @param _from address of tokens sender
+     * @param _value requsted amount of bridged tokens
+     * @param _data alternative receiver, if specified
+     */
+    function bridgeSpecificActionsOnTokenTransfer(
+        ERC677, /* _token */
+        address _from,
+        uint256 _value,
+        bytes _data
+    ) internal {
+        if (!lock()) {
+            passMessage(chooseReceiver(_from, _data), _value);
+        }
+    }
+
+    /**
+     * @dev Executes action on relayed request to fix the failed transfer of tokens
+     * @param _recipient address of tokens receiver
+     * @param _value amount of fixed tokens
+     */
+    function executeActionOnFixedTokens(address _recipient, uint256 _value) internal {
+        _transferWithOptionalMint(_recipient, _value);
+    }
+
+    /**
+     * @dev Internal function for transfer of tokens, with optional minting if current balance is insufficient
+     * @param _recipient address of tokens receiver
+     * @param _value amount of fixed tokens
+     */
+    function _transferWithOptionalMint(address _recipient, uint256 _value) internal {
+        IBurnableMintableERC677Token token = IBurnableMintableERC677Token(erc677token());
+        uint256 balance = token.balanceOf(address(this));
+        if (_recipient != address(0) && balance == 0) {
+            token.mint(_recipient, _value);
+        } else if (balance < _value) {
+            token.mint(address(this), _value - balance);
+            token.transfer(_recipient, _value);
```

Comment:
Please note that the `transfer` method doesn't allow the `_recipient` to be `address(0)` as well as the `mint` (see the deployed token contract on Mainnet: https://etherscan.io/address/0x0ae055097c6d159879521c384f1d2123d1f195e6#code). I guess the bridge should restrict setting the `recipient` to zero on the `Home` side (in case of using the `alternative receiver` feature). If some user wants to burn their tokens through the bridge (sending them from Home to Foreign and setting the recipient as zero), they can use another non-zero address (like [0x00...01](https://etherscan.io/address/0x0000000000000000000000000000000000000001)).

Reply:
This restriction is already included in the `amb-erc-to-erc`
https://github.com/poanetwork/tokenbridge-contracts/blob/393ef6c10bb59d1d370b4de9df77c23c29ac7f0f/contracts/upgradeable_contracts/BaseERC677Bridge.sol#L29-L37
So, calls like `token.transferAndCall(mediator, value, 0x00..00)` will be reverted

Reply:
Ok 👍

Reply:
Then maybe `_recipient != address(0)` condition is redundant here? If the recipient can never be zero due to that condition in `chooseReceiver`.

Reply:
Yes, you are right. It is redundant here

Reply:
The condition is about to be removed in https://github.com/poanetwork/tokenbridge-contracts/pull/408